### DISCUSSION TITLE
Task 1: How to change Checkpoints folder location

### DISCUSSION THREAD

**Discussion by dskhanirfan** [FETS-AI/Challenge#91] [General]
How can I change the Checkpoints location as they are stored in ~/.local/workspace/checkpoint
My .local disk space on the cluster is limited so I want to set the location of checkpoints folder to the **Challenge/Task_1** folder

**Comment by sarthakpati** [comment]
I _believe_ you can edit the `checkpoint_folder` location in the [`setup_checkpoint_folder`](https://github.com/FETS-AI/Challenge/blob/main/Task_1/fets_challenge/checkpoint_utils.py#L20) function.

**WARNING**: this customized function will **not** be part of the submission.

Tagging @alexey-gruzdev and @psfoley if I am missing something?

**Reply by dskhanirfan** [reply]
Thanks it worked

**Reply by psfoley** [reply]
@dskhanirfan There are actually a few references in the code that would need to be changed in order for the checkpointing logic to work correctly beyond what @sarthakpati mentioned. What I might actually recommend instead as a workaround is to create a symbolic link from `~/.local/workspace/checkpoint/` to a different disk where you have more space

**Reply by sarthakpati** [reply]
>  create a symbolic link from `~/.local/workspace/checkpoint/` to a different disk

Definitely a better solution, thank you @psfoley!

**Reply by dskhanirfan** [reply]
Although I made symbolic link for .local/workspace/checkpoint/ I did 
ln -s .local/workspace/checkpoint Target_folder
still my local memory is used which is not permitted.

**Reply by sarthakpati** [reply]
You will need to do the following: 

```bash
ln -s ~/.local/workspace/checkpoint /path/to/target
```

**Reply by dskhanirfan** [reply]
I did the same but my local drive is still getting full

**Reply by sarthakpati** [reply]
Can you confirm if it is correctly defined? Could you post the output of the following command: `ls -lt ~/.local/workspace/checkpoint`

**Reply by dskhanirfan** [reply]
Output
total 24
drwxrws---. 2 irfan21h irfan21h 4096 Jul 14 14:22 experiment_6
drwxrws---. 2 irfan21h irfan21h 4096 Jul 14 13:20 experiment_5
drwxrws---. 2 irfan21h irfan21h 4096 Jul 14 12:10 experiment_4
drwxrws---. 2 irfan21h irfan21h 4096 Jul 14 11:55 experiment_3
drwxrws---. 2 irfan21h irfan21h 4096 Jul 14 10:29 experiment_2
drwxrws---. 2 irfan21h irfan21h 4096 Jul 14 09:58 experiment_1

**Reply by sarthakpati** [reply]
Actually, how about this: `ls -lt ~/.local/workspace/check*`

**Reply by dskhanirfan** [reply]
Same output with ls -lt ~/.local/workspace/check*

**Reply by sarthakpati** [reply]
What about `ls -lt ~/.local/workspace`

**Reply by dskhanirfan** [reply]
(base) [irfan21h@puhti-login2 Task_1]$ ls -lt ~/.local/workspace
total 264
-rw-rw----. 1 irfan21h irfan21h    320 Jul 14 19:05 temp_colab.npy
drwxrws---. 8 irfan21h irfan21h   4096 Jul 14 13:13 checkpoint
-rw-rw----. 1 irfan21h irfan21h   3613 Jul 14 13:13 agg_to_col_one_signed_cert.zip
-rw-rw----. 1 irfan21h irfan21h   3614 Jul 14 13:13 agg_to_col_two_signed_cert.zip
-rw-rw----. 1 irfan21h irfan21h 189135 Jul 14 13:13 gandlf_paths.csv
drwxrws---. 3 irfan21h irfan21h   4096 Jul 14 06:40 impl
-rwxrwx--x. 1 irfan21h irfan21h   8207 Jul 14 06:40 partitioning_1.csv
-rwxrwx--x. 1 irfan21h irfan21h   8207 Jul 14 06:40 partitioning_2.csv
drwxrws---. 3 irfan21h irfan21h   4096 Jul 14 06:40 plan
-rw-rw----. 1 irfan21h irfan21h     18 Jul 14 06:40 requirements.txt
-rw-rw----. 1 irfan21h irfan21h    288 Jul 14 06:40 small_split.csv
drwxrwx---. 2 irfan21h irfan21h   4096 Jun  3 23:45 save
drwxrwx---. 5 irfan21h irfan21h   4096 Jun  3 15:38 cert
drwxrwx---. 2 irfan21h irfan21h   4096 Jun  3 15:37 code
drwxrwx---. 2 irfan21h irfan21h   4096 Jun  3 15:37 data
drwxrwx---. 2 irfan21h irfan21h   4096 Jun  3 15:37 logs

**Reply by sarthakpati** [reply]
So, this shows that the symlink was not created properly.

This was my bad.

> You will need to do the following:
> 
> ```shell
> ln -s ~/.local/workspace/checkpoint /path/to/target
> ```

This should have been:

```bash
mv ~/.local/workspace/checkpoint ~/.local/workspace/checkpoint_old # saving previous results
ln -s /path/to/target ~/.local/workspace/checkpoint 
```

**Reply by dskhanirfan** [reply]
mv ~/.local/workspace/checkpoint ~/.local/workspace/checkpoint_old
mkdir blackhole
ln -s /scratch/project_2004598/PLAYGROUND/blackhole ~/.local/workspace/checkpoint

 ls -lt ~/.local/workspace
total 264
lrwxrwxrwx. 1 irfan21h irfan21h     45 Jul 14 19:34 checkpoint -> /scratch/project_2004598/PLAYGROUND/blackhole
drwxrws---. 8 irfan21h irfan21h   4096 Jul 14 19:30 checkpoint_old
-rw-rw----. 1 irfan21h irfan21h    320 Jul 14 19:28 temp_colab.npy
-rw-rw----. 1 irfan21h irfan21h   3613 Jul 14 13:13 agg_to_col_one_signed_cert.zip
-rw-rw----. 1 irfan21h irfan21h   3614 Jul 14 13:13 agg_to_col_two_signed_cert.zip
-rw-rw----. 1 irfan21h irfan21h 189135 Jul 14 13:13 gandlf_paths.csv
drwxrws---. 3 irfan21h irfan21h   4096 Jul 14 06:40 impl
-rwxrwx--x. 1 irfan21h irfan21h   8207 Jul 14 06:40 partitioning_1.csv
-rwxrwx--x. 1 irfan21h irfan21h   8207 Jul 14 06:40 partitioning_2.csv
drwxrws---. 3 irfan21h irfan21h   4096 Jul 14 06:40 plan
-rw-rw----. 1 irfan21h irfan21h     18 Jul 14 06:40 requirements.txt
-rw-rw----. 1 irfan21h irfan21h    288 Jul 14 06:40 small_split.csv
drwxrwx---. 2 irfan21h irfan21h   4096 Jun  3 23:45 save
drwxrwx---. 5 irfan21h irfan21h   4096 Jun  3 15:38 cert
drwxrwx---. 2 irfan21h irfan21h   4096 Jun  3 15:37 code
drwxrwx---. 2 irfan21h irfan21h   4096 Jun  3 15:37 data
drwxrwx---. 2 irfan21h irfan21h   4096 Jun  3 15:37 logs

**Reply by sarthakpati** [reply]
This looks good:
```
ls -lt ~/.local/workspace
total 264
lrwxrwxrwx. 1 irfan21h irfan21h 45 Jul 14 19:34 checkpoint -> /scratch/project_2004598/PLAYGROUND/blackhole
```

**Reply by dskhanirfan** [reply]
I get the following error

```python-traceback
Traceback (most recent call last):
  File "FeTS_Challenge.py", line 26, in <module>
srun: error: r04g05: task 0: Exited with exit code 1
srun: launch/slurm: _step_signal: Terminating StepId=6906869.0
    from fets_challenge import run_challenge_experiment
  File "/scratch/project_2004598/14JULY/Challenge/Task_1/fets_challenge/__init__.py", line 6, in <module>
    from .experiment import run_challenge_experiment
  File "/scratch/project_2004598/14JULY/Challenge/Task_1/fets_challenge/experiment.py", line 345
    logger.info(f'\nCreated experiment folder {checkpoint_folder}...')
                                                                    ^
SyntaxError: invalid syntax
```

**Reply by sarthakpati** [reply]
Can you try putting a `print(checkpoint_folder)` before this line in `experiment.py`, reinstall the notebook dependencies, and try running again? Please put the output of this print statement.

**Reply by dskhanirfan** [reply]
I reinstalled the repository and followed all the instructions and I have symlink enabled. Thanks
